pinball/authentication/oauth2.py

```diff
@@ -257,7 +258,7 @@ def process_request(self, request):
                 domain = crypter.decrypt(request.COOKIES['domain_url'])
             except CryptoException:
                 return HttpResponseRedirect('/signin/')
-            if user_id == request.COOKIES['user_id'] and user_id != '' \
+            if _hmac.compare_digest(user_id, request.COOKIES['user_id']) and user_id != '' \
```

Comment:
Because the user_id is coming from an authenticated/encrypted cookie (it can only be set by the server), I don't think there is the possibility for a timing attack here. Please let me know if I missed something.